python/src/iceberg/catalog/base.py

```diff
@@ -0,0 +1,198 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from iceberg.schema import Schema
+from iceberg.table.base import PartitionSpec, Table
+
+
+class Catalog(ABC):
+    """
+    Base Catalog for table operations like - create, drop, load, list and others.
+
+    Attributes:
+        name(str): Name of the catalog
+        properties(dict): Catalog properties
+    """
+
+    def __init__(self, name: str, properties: dict):
+        self._name = name
+        self._properties = properties
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def properties(self) -> dict:
+        return self._properties
+
+    @abstractmethod
+    def list_tables(self) -> list:
```

Comment:
```suggestion
    def list_tables(self) -> list[Table]:
```

Reply:
Thanks for the suggestion Fokko! Although the built-in collection typing, as suggested above, only works in Python 3.9 and above. But I see where you are going with this, so let me return a `typing.List` instead. Ex: `List[Table]`

Reply:
Added collection typing for all catalog APIs now. Thanks!